File: datastructure/arrays/easy/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for num, value in enumerate(nums, start=0):
            valuetofind = target - value
            indextofind = None
            for subnum, subvalue in enumerate(nums, start=0):
                if subnum == num:
                    continue
                if subvalue == valuetofind:
                    indextofind = subnum
                    break
            if indextofind != None:
                return [num, indextofind]


car = Solution()
print(car.twoSum([2, 11, 15, 7], 9))

File: datastructure/arrays/easy/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
/**
 * Runtime: 192 ms
 * Memory: 42 MB
 */
const twoSum = function (nums, target) {
  let foundIndices = false;
  let indices = [];
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] === target) {
        foundIndices = true;
        indices.push(i,j);
      }
      if (foundIndices) {
        break;
      }
    }
    if (foundIndices) {
      break;
    }
  }
  return indices;
};

console.log(twoSum([3,2,4], 6))